src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -1451,7 +1463,15 @@ def __list_fill__(cls, parent, name, data, options=None):
             data_shape = (len(data),)
         else:
             data_shape = get_data_shape(data)
-
+        if expandable:
+            if 'maxshape' not in io_settings:
+                io_settings['maxshape'] = tuple([None]*len(data_shape))
+            else:
+                # Don't override existing settings
+                pass
+        else:
+            msg = "Datasets written using user defined parameters. Default expandable via chunking: False"
+            warnings.warn(msg)
```

Comment:
```suggestion
```

I don't think the `warning` is necessary, because it is only being raised when the user explicitly sets `expandable=False`, so the user explicitly made that choice. The `else: pass` part I don't think is needed either.

Reply:
So the pass, was just for me to be explicit in my conditionals. I can remove it, but I think the note provides clarity for the future. 

The warning I think is a good reminder for now since this is a change to a default. Maybe we can remove it after this feature has been here for a while, but I think having some "flashing lights" is helpful.

I can remove these if you feel strongly about it.

Reply:
I'll remove the pass, but will keep the comment

Reply:
I do like the warning still

Reply:
The large number of warnings for a single file is a good point. I'll make the changes and ping for review to merge.